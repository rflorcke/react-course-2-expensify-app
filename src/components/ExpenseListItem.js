import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`}><h1>{description}</h1></Link>
        <h2>${amount / 100}</h2>
        <p>Created at: {createdAt}</p>
    </div>
);

export default ExpenseListItem;