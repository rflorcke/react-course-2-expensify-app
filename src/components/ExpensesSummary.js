import React from "react";
import { connect } from "react-redux";
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";
import numeral from "numeral";

export class ExpensesSummary extends React.Component {
    renderMessage() {
        if (this.props.countExpenses > 0) {
            const { countExpenses: count, totalAmount } = this.props;
            const total = numeral(totalAmount / 100).format("$0,0.00");
            const msg = `Viewing ${count} ${count === 1 ? "expense" : "expenses"} totalling ${total}`;
            return (
                <h2>{msg}</h2>
            );
        }
        return null;
    }
    render() {
        return (
            <div>{this.renderMessage()}</div>
        );
    }
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        countExpenses: visibleExpenses.length,
        totalAmount: selectExpensesTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);