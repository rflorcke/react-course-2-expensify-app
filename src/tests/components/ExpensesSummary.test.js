import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";
import expenses from "../fixtures/expenses";

test("should correctly render ExpensesSummary with 1 expense", () => {
    const wrapper = shallow(<ExpensesSummary countExpenses={1} totalAmount={235} />);
    expect(wrapper).toMatchSnapshot();
});

test("should correctly render ExpensesSummary with multiple expense", () => {
    const wrapper = shallow(<ExpensesSummary countExpenses={5} totalAmount={503500} />);
    expect(wrapper).toMatchSnapshot();
});