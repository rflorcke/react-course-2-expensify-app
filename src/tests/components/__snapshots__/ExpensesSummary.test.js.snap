// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h2>
    Viewing 1 expense totalling $2.35
  </h2>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expense 1`] = `
<div>
  <h2>
    Viewing 5 expenses totalling $5,035.00
  </h2>
</div>
`;
